src/formatter-settings/utils.ts

```diff
@@ -39,22 +42,28 @@ export function isRemote(path: string): boolean {
 
 export async function addDefaultProfile(context: vscode.ExtensionContext): Promise<void> {
     const defaultProfile: string = path.join(context.extensionPath, "webview-resources", "java-formatter.xml");
-    const targetFileName = "java-formatter.xml";
-    let profilePath: string;
+    const targetPath = await getTargetPath(context, "java-formatter.xml");
+    const profilePath = targetPath.profilePath;
+    await fse.copy(defaultProfile, profilePath);
+    const workspaceFolders = vscode.workspace.workspaceFolders;
+    await vscode.workspace.getConfiguration("java").update("format.settings.url", (workspaceFolders?.length ? targetPath.relativePath : profilePath), !(workspaceFolders?.length));
+    vscode.commands.executeCommand("vscode.openWith", vscode.Uri.file(profilePath), "java.formatterSettingsEditor");
+}
+
+export async function getTargetPath(context: vscode.ExtensionContext, fileName: string): Promise<{relativePath: string, profilePath: string}> {
     const workspaceFolders = vscode.workspace.workspaceFolders;
+    let profilePath: string;
     if (workspaceFolders?.length) {
-        profilePath = path.posix.join(workspaceFolders[0].uri.fsPath, ".vscode", targetFileName);
+        profilePath = path.join(workspaceFolders[0].uri.fsPath, ".vscode", fileName);
     } else {
         const folder: string = context.globalStorageUri.fsPath;
         await fse.ensureDir(folder);
-        profilePath = path.posix.join(folder, targetFileName);
+        profilePath = path.join(folder, fileName);
     }
-    // bug: https://github.com/redhat-developer/vscode-java/issues/1944, only the profiles in posix path can be monitored when changes, so we use posix path for default profile creation temporarily.
-    const relativePath = toPosixPath(path.join(".vscode", targetFileName));
+     // bug: https://github.com/redhat-developer/vscode-java/issues/1944, only the profiles in posix path can be monitored when changes, so we use posix path for default profile creation temporarily.
+    const relativePath = toPosixPath(path.join(".vscode", fileName));
     profilePath = toPosixPath(profilePath);
-    await fse.copy(defaultProfile, profilePath);
-    await vscode.workspace.getConfiguration("java").update("format.settings.url", (workspaceFolders?.length ? relativePath : profilePath), !(workspaceFolders?.length));
-    vscode.commands.executeCommand("vscode.openWith", vscode.Uri.file(profilePath), "java.formatterSettingsEditor");
+    return {relativePath, profilePath};
```

Comment:
Can you use `workspace.asRelativePath()` to get the relative path instead of returning a object?